tests/experimental/datasources/test_pandas_s3_datasource.py

```diff
@@ -0,0 +1,267 @@
+from great_expectations.experimental.datasources.file_path_data_asset import (
+    _FilePathDataAsset,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.pandas_file_path_datasource import (
+    CSVAsset,
+)
+
+logger = logging.getLogger(__file__)
+
+
+try:
+    import boto3
+    import botocore
+    from botocore.client import BaseClient
+except ImportError:
+    logger.debug(
+        "Unable to load boto3 or botocore; install optional boto3 and botocore dependencies for support."
+    )
+    boto3 = None
+    botocore = None
+    BaseClient = None
+
+
+# apply markers to entire test module
+pytestmark = [
+    pytest.mark.skipif(
+        PANDAS_VERSION < 1.2, reason=f"ZEP pandas not supported on {PANDAS_VERSION}"
+    )
+]
+
+
+def _get_region_name() -> str:
+    return "us-east-1"
+
+
+def _get_boto3_client() -> BaseClient:
+    client = boto3.client("s3", region_name=_get_region_name())
+    return client
+
+
+def _build_pandas_s3_datasource() -> PandasS3Datasource:
+    bucket: str = "test_bucket"
+    conn = boto3.resource("s3", region_name=_get_region_name())
+    conn.create_bucket(Bucket=bucket)
+    client = _get_boto3_client()
+
+    test_df: pd.DataFrame = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+
+    keys: List[str] = [
+        "alex_20200809_1000.csv",
+        "eugene_20200809_1500.csv",
+        "james_20200811_1009.csv",
+        "abe_20200809_1040.csv",
+        "will_20200809_1002.csv",
+        "james_20200713_1567.csv",
+        "eugene_20201129_1900.csv",
+        "will_20200810_1001.csv",
+        "james_20200810_1003.csv",
+        "alex_20200819_1300.csv",
+    ]
+    for key in keys:
+        client.put_object(
+            Bucket=bucket, Body=test_df.to_csv(index=False).encode("utf-8"), Key=key
+        )
+
+    return PandasS3Datasource(
+        name="pandas_s3_datasource",
+        bucket=bucket,
+    )
+
+
+def _build_csv_asset() -> _FilePathDataAsset:
+    pandas_s3_datasource: PandasS3Datasource = _build_pandas_s3_datasource()
+    asset = pandas_s3_datasource.add_csv_asset(
+        name="csv_asset",
+        regex=r"(?P<name>.+)_(?P<timestamp>.+)_(?P<price>\d{4})\.csv",
+    )
+    return asset
+
+
+def _build_bad_regex_config() -> tuple[re.Pattern, TestConnectionError]:
+    asset = _build_csv_asset()
+    regex = re.compile(
+        r"(?P<name>.+)_(?P<ssn>\d{9})_(?P<timestamp>.+)_(?P<price>\d{4})\.csv"
+    )
+    data_connector: S3DataConnector = cast(S3DataConnector, asset._data_connector)
+    test_connection_error = TestConnectionError(
+        f"""No file in bucket "{asset.datasource.bucket}" with prefix "{data_connector._prefix}" matched regular expressions pattern "{regex.pattern}" using deliiter "{data_connector._delimiter}" for DataAsset "{asset.name}"."""
+    )
+    return regex, test_connection_error
+
+
+@pytest.mark.integration
```

Comment:
Please note that all tests are marked "integration" here, because of the "S3" access (albeit a "mock" one).  Thanks.